Add an article on how to use Ansible and the Yaml Config to automate Wildfly's subsystems config.

rpelisse opened the pull request:
@bstansberry Let me know what you think about this proposal!

rpelisse commented:
@bstansberry Any chance to get this article approved and release soon? Anything I can change to make it better?

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+* Large alteration of the subsystems (adding one or several, or simply rewriting entirely the default configuration) are most likely easier to achieve by providing a modified base configuration.
+* Small changes to the subsystem configuration, adding a few, straightforward resources are most likely easy enough to implement.
+* Changes in the configuration linked to the target environments can be achieved using the templating feature of Ansible.
+* No matter what, remember the https://en.wikipedia.org/wiki/KISS_principle[KISS principe (Keep It Stupid Simple)].
```
s/principe/principle

You're nice; for us mean unprofessional people it's "Keep It Simple, Stupid!"  Your way is better for this kind of article. :)

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+* Changes in the configuration linked to the target environments can be achieved using the templating feature of Ansible.
+* No matter what, remember the https://en.wikipedia.org/wiki/KISS_principle[KISS principe (Keep It Stupid Simple)].
+
+Let's run again the `playbook`. As in the above example run, Ansible will notice the changed to the Yaml configuration file and consequently update the target's subsystems configuration, before restarting the server.
```
s/changed/change

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+
+=== Deploy JDBC drivers and datasources
+
+The deployment of JDBC drivers and datasources on the target system is a somewhat more elaborated use case than the one we just saw with the messaging subsystem. Indeed, to add a JDBC driver to a WildFly server an entire module must be created, it's not just a configuration change in the `standalone.xml` that needs to be performed in an idempotent manner.
```
s/created,/created;/g

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+
+The deployment of JDBC drivers and datasources on the target system is a somewhat more elaborated use case than the one we just saw with the messaging subsystem. Indeed, to add a JDBC driver to a WildFly server an entire module must be created, it's not just a configuration change in the `standalone.xml` that needs to be performed in an idempotent manner.
+
+Fortunately, here again, the Ansible collection for WildFly does most of the heavy lifting. In fact, the default `playbook` we used already comes with the setup of two JDCB drivers:
```
s/JDCB/JDBC

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+...
+----
+
+As shown above, the collection provides a generic role that takes care of creating the file hierarchy associated to a JDBC driver, but also download the required artifacts (jar file) along with generating the needed descriptor (module.xml). The driver's specific values are stored in the `vars.yml`, imported by Ansible when executing this `playbook`:
```
s/download/downloading

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+    - { version: "{{ mariadb_driver_version }}", name: 'org.mariadb', jar_file: "mariadb-java-client-{{ mariadb_driver_version }}.jar", url: "https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/{{ mariadb_driver_version }}/mariadb-java-client-{{ mariadb_driver_version }}.jar" }
+----
+
+*Note:* _The Ansible collection for WildFly comes with a default template to generate the `module.xml` of a custom module. Obviously, this template might not be a good fit for ALL the drivers that user may have to install in a WildFly setup. For this reason, the template itself can easily be replaced by another one, providing by the user._
```
s/user/users
s/providing/provided

bstansberry commented on _posts/2024-07-10-ansible-yaml-wildfly.adoc:
```diff
@@ -0,0 +1,740 @@
+
+*Note:* _The Ansible collection for WildFly comes with a default template to generate the `module.xml` of a custom module. Obviously, this template might not be a good fit for ALL the drivers that user may have to install in a WildFly setup. For this reason, the template itself can easily be replaced by another one, providing by the user._
+
+If this role ensures the modules are ready to be utilized, it does not, however, activate them. To make them available to use for datasources, we will add their definition to our Yaml configuration file:
```
s/If this/While this/g

bstansberry reviewed (changes requested):
@rpelisse This is good stuff. Once published WDYT about converting this to guides for https://www.wildfly.org/guides/?

rpelisse commented:
@bstansberry Sorry for the delay, I have been on sick leave for the past weeks.

I've added all your fixes (check the new commit if you want to double check). Assuming it's good for you, I can squash the 4 commits into one and we can (hopefully) release this tomorrow? (I've updated the date accordingly).

Regarding the guide, why not? I can certainly took a look at updating the existing one with some of this content.

bstansberry commented:
Thanks @rpelisse! Very nice article!
